### ISSUE TITLE
Mini.files gives error and crashes on closing and reopening the Files windows

### ISSUE THREAD

**Issue by SwayKh** [echasnovski/mini.nvim#1419]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

Mini.files

### Description

When I open Mini.Files, and close the window, reopening mini.files gives an error and causes mini.files to stop working.

I have simple mini.files config, and one keymap for it.
```lua
  require("mini.files").setup({
    mappings = {
      go_in_plus = "<cr>",
      synchronize = "<c-s>",
    },
    windows = {
      max_number = 3,
      preview = true,
    },
  })

  vim.keymap.set("n", "-", function()
    MiniFiles.open(vim.api.nvim_buf_get_name(0))
    MiniFiles.reveal_cwd()
  end, { desc = "Open Mini Files" })
```

Some other things I observed.
1. After Mini.files crashes, the "-" keybind stops working.
2. If I navigate to some other files, Using Mini.Pick or Arrow.nvim, and press "-" again. Mini.Files opens, but closing that window and reopening it again give me the same error
3. If I replace the `MiniFiles.Open()` call with `MiniFiles.open(vim.api.nvim_buf_get_name(0), false)` like given in the readme, to open a "fresh" window, that makes everything work as expected, and I can close and reopen MiniFiles as many times as I want.

### Neovim version

NVIM v0.10.2

### Steps to reproduce

1. `nvim -nu minimal.lua`
2. Press "-" to open Mini.Files window
3. Press q to exit the window
4. Press "-" to open Mini.files window again

### Expected behavior

Reopen the Mini.Files window again...

### Actual behavior

I receive this error
```
Error executing vim.schedule lua callback: ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:2378: attempt to compare number with nil
stack traceback:
        ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:2378: in function 'is_modified_buffer'
        ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:2550: in function 'window_update_border_hl'
        ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:2519: in function 'window_set_view'
        ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:1779: in function 'explorer_refresh_depth_window'
        ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:1505: in function 'explorer_refresh'
        ...e/nvim/site/pack/deps/start/mini.nvim/lua/mini/files.lua:778: in function 'open'
        /home/sumit/.config/nvim/lua/plugins/mini.lua:61: in function </home/sumit/.config/nvim/lua/plugins/mini.lua:60>
```

**Comment by echasnovski** [comment]
Thanks for the issue!

I can not reproduce with *only* 'mini.files' enabled with config from "Description" (i.e. no other plugins or mappings). There are no errors after several `-`, `q`, `-`, `q` cycles. Would you mind trying to reproduce this locally with only 'mini.files' enabled?

**Comment by SwayKh** [comment]
Okay, I tried to make a minimal.lua file, And That doesn't seem to have this issue. So I don't know what the issue might be.
```lua
vim.loader.enable()

-- Clone 'mini.nvim' manually in a way that it gets managed by 'mini.deps'
local path_package = vim.fn.stdpath("data") .. "/site/"
local mini_path = path_package .. "pack/deps/start/mini.nvim"
if not vim.loop.fs_stat(mini_path) then
	vim.cmd('echo "Installing `mini.nvim`" | redraw')
	local clone_cmd = {
		"git",
		"clone",
		"--filter=blob:none",
		"https://github.com/echasnovski/mini.nvim",
		mini_path,
	}
	vim.fn.system(clone_cmd)
	vim.cmd("packadd mini.nvim | helptags ALL")
	vim.cmd('echo "Installed `mini.nvim`" | redraw')
end

-- Set up 'mini.deps' (customize to your liking)
require("mini.deps").setup({ path = { package = path_package } })

require("mini.files").setup({
	mappings = {
		go_in_plus = "<cr>",
		synchronize = "<c-s>",
	},
	windows = {
		max_number = 3,
		preview = true,
	},
})

vim.keymap.set("n", "-", function()
	MiniFiles.open(vim.api.nvim_buf_get_name(0))
	MiniFiles.reveal_cwd()
end, { desc = "Open Mini Files" })
```

**Comment by SwayKh** [comment]
Actually, My MiniJump2D is also a bit broken for the past few days, I didn't bother trying to fix that(Or thought anything of it when I noticed), Since I don't use it that much. But That's also not occurring with the minimal.lua example.

In case of MiniJump2D, I have to press the "label letters" sequence twice to jump to the place I want. For example,
1. I press <CR> to start MiniJump2D
2. Press, let's say "r' and then "w" labels
3. after pressing "w", the label resets to original un-filtered labels, And I have to press "r" and "w" again to jump.

**Comment by echasnovski** [comment]
> So I don't know what the issue might be.

My guess would be to start with the following:
- Execute `:nmap q` when inside 'mini.files' buffer. Maybe it gets defined to do something else.
- Try disabling plugins which might interact with floating windows or key presses (like 'folke/which-key.nvim').
- Fully bisect config.

> Actually, My MiniJump2D is also a bit broken for the past few days, I didn't bother trying to fix that(Or thought anything of it when I noticed), Since I don't use it that much. But That's also not occurring with the minimal.lua example.

Then I am afraid it is definitely some third party plugin (or you config function/autocommand) behaving in conflicting manner.

Closing as not an issue with 'mini.nvim' per se. If you find a reproducible example which uses only it, please let me know.

**Comment by SwayKh** [comment]
Thanks, for the tips on how to pinpoint the issue, That worked wonders, I figured out the cause.

It's the plugin [hardtime.nvim](https://github.com/m4xshen/hardtime.nvim). I recently added it to my config.
Disabling this plugin fixes both the MiniFiles and MiniJump2D issues.

Still quite weird that hardtime somehow causes issues with mini.nvim

This was my hardtime config
```lua
local function hardtime()
  add({
    source = "m4xshen/hardtime.nvim",
    depends = {
      "MunifTanjim/nui.nvim",
    },
  })
  require("hardtime").setup()
end
```
